pandas/core/generic.py

```diff
@@ -1109,10 +1109,20 @@ def to_json(self, path_or_buf=None, orient=None, date_format='epoch',
               - index : dict like {index -> {column -> value}}
               - columns : dict like {column -> {index -> value}}
               - values : just the values array
+            - ``'table_schema'`` dict like
+              ``{'schema': [schema], 'data': [data]}``
+              where the schema component comes from
+              ``build_table_schema`` and the data component is
+              like ``orient='records'``.
+
+            .. versionchanged:: 0.20.0
+            Added the ``'table_schema'`` orient.
 
         date_format : {'epoch', 'iso'}
             Type of date conversion. `epoch` = epoch milliseconds,
-            `iso`` = ISO8601, default is epoch.
+            `iso` = ISO8601. Default is epoch, except when orient is
```

Comment:
Should adding a `timedelta_format`, distinct from `date_format` wait until #15137 is fixed?
Right now, I don't know what `timedelta_format` would do for any of the other json orients. Unless you want me to move my timedelta hack out of the `JSONTableWriter` to the base writer, and have it apply to all orients.